### PR TITLE
[MKC-1832] MicroPython update examples with GIGA exception

### PR DESCRIPTION
**Pull request by pedromsousalima**
## What This PR Changes
- Added exception for MicroPython when using GIGA boards

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/micropython/01.basics/05.digital-analog-pins/digital-analog-pins.md [hunk 1]
```diff
@@ -134,3 +136,34 @@ pwm.freq(1000)
...
+ch.pulse_width_percent(25)
+```
+
+In this example:
+
+- `p` is a `Pin` object initialized on pin `'A13'`. This pin is set up for PWM output.
+- `tim` is a `Timer` object initialized with timer number 2 and a frequency of 1000 Hz.
+- `ch` is a PWM channel created on timer 2, channel 1, and is associated with pin `p`.
+- `ch.pulse_width_percent(25)` sets the duty cycle of the PWM signal to 25%.
+- `'A13'` is the pin markedas **DAC1**
```
Comment (karlsoderby) [inline content/micropython/01.basics/05.digital-analog-pins/digital-analog-pins.md:165]:
```suggestion
- `'A13'` is marked as **DAC1** on the GIGA R1 WiFi

```

File: content/micropython/01.basics/05.digital-analog-pins/digital-analog-pins.md [hunk 2]
```diff
@@ -134,3 +136,34 @@ pwm.freq(1000)
...
 ```
+In this example:
+
+- `Pin(15)` is attached to PWM
+- `duty` is set at 3000
+- `freq` is set at 1 Hz
+
+### PWM on the GIGA R1
+
+On STM32 boards like the Arduino GIGA R1, PWM is handled differently. You need to use the `Timer` class along with the `Pin` class from the `pyb` module.
```
Comment (karlsoderby) [inline content/micropython/01.basics/05.digital-analog-pins/digital-analog-pins.md:147]:
```suggestion
On STM32 boards like the [Arduino GIGA R1 WiFi](https://store.arduino.cc/products/giga-r1-wifi), PWM is handled differently. You need to use the `Timer` class along with the `Pin` class from the `pyb` module.

```

**Review by karlsoderby — CHANGES_REQUESTED** [review]
Some minor changes @pedromsousalima , otherwise looking good!

**Comment by pedromsousalima** [comment]
@karlsoderby applied changes thanks!

**Review by karlsoderby — APPROVED** [review]

**Pull request merged**